akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -119,7 +125,7 @@ private[io] object SelectionHandler {
 
     private[this] val select = new Task {
       def tryRun(): Unit = {
-        if (selector.select() > 0) { // This assumes select return value == selectedKeys.size
+        if (selector.select(MaxSelectMillis) > 0) { // This assumes select return value == selectedKeys.size
```

Comment:
Yes, our code doesn't do anything in particular. It's selector internals which seem to cleanup things on reissuing select I guess.